app/workers/start_of_cycle_notification_worker.rb

```diff
@@ -0,0 +1,73 @@
+class StartOfCycleNotificationWorker
+  include Sidekiq::Worker
+
+  def perform(service, batch_size = 100)
+    @service = service
+
+    providers_scope.find_each(batch_size: batch_size) do |provider|
+      provider.provider_users.where.not(Arel.sql(chaser_sent_sql)).each do |provider_user|
+        ProviderMailer.send(mailer_method, provider_user)
+        ChaserSent.create!(chased: provider_user, chaser_type: mailer_method)
+
+        next if service == :apply
```

Comment:
maybe we could tidy up the statements we check around into a method or two to say for instance `provider_user_has_organisation_permissions_and_a_pending_relationship?` etc.